### DISCUSSION TITLE
Pooled Mean Matching MICE

### DISCUSSION THREAD

**Discussion by MikeDereviankin** [amices/mice#524] [Missing data methodology]
For MICE imputations I need to constrict the predictions so that the predicted values will have the same mean (which is a measured value). The situation is we are dealing with mean blood serum samples (individual blood samples are pooled together) where we have measured values, which are representative of the mean of those individuals. I am trying to predict what the concentration of x was in those individuals based on the measured mean and covariate data. You'll notice in my dummy dataset that there are 3 individuals (Individual_id) for each pool (Pool_id). So when imputing these values to the individuals we need the average of those 3 individuals to equal the Pool_mean.

How can we constrict the Mice algorithm to still predict based on covariate data, but have the means match exactly (can be any method chose, "cart", in this circumstance)? Could this conceptually be done through a MICE squeeze constraint with inputs from the mean?

The code is below:

```
library(mice)
library(dplyr)

#create demo data table as an example

Pool_id <- c(1, 1, 1, 2, 2, 2, 3, 3, 3)
Pool_mean <- c(15, 15, 15, 35, 35, 35, 42, 42, 42)
Individual_id <- c(1, 2, 3, 4, 5, 6, 7, 8, 9)
concentration <- c(10, 20, NA, 30, NA,NA, NA, NA, 70)
co_variate <- c(0.1, 0.2, 0.1, 0.2, 0.3, 0.1, 0.1, 0.2, 0.3)

df <- data.frame(Pool_id, Pool_mean, Individual_id, concentration, co_variate)

#run mice to impute missing data

initial_imputed <- mice(df, m = 5, maxit = 10, meth = "cart", seed = 3985))
completed <- complete(intial_imputed)
```

I know that we can constraint mice using a post process and maybe a custom function like the vec_squeeze below. However, I need to constraint the values based on a mean. How could I update this function to create this?

`vec_squeeze <- function(x, bounds) {
  stopifnot(length(x) == nrow(bounds))
  pmin(pmax(x, bounds[,1]), bounds[,2])
}`

**Comment by stefvanbuuren** [comment]
Chapter 5 in <https://dspace.library.uu.nl/bitstream/handle/1874/308699/vink.pdf> studies a related problem. Vink proposes predictive ratio matching to divide the remainder of a sum. If it is sensible to organise your thee observations per person within one row, then you probably could use Vink's approach right away. Be aware though that there is no procedure in `mice` that does that actual calculations, so you will need to write your own software.

If it is not sensible to re-organise (e.g. when different persons have different numbers of observations) I don't think there is a good solution.
